### ISSUE TITLE
Error: error getting synthetics test:  is not a valid SyntheticsDeviceID

### ISSUE THREAD

**Issue by mjhuber** [terraform-providers/terraform-provider-datadog#538]
This is only affecting us with provider version 2.8.0.  It works fine with v2.7.0.

### Terraform Version
Terraform v0.12.25

### Affected Resource(s)
- datadog_synthetics_test

### Terraform Configuration Files
```hcl
resource "datadog_synthetics_test" "homepage_browser" {
  for_each = { for monitor in local.monitors : "${monitor.id}" => monitor }

  type    = "browser"
  status  = each.value.status
  name    = "homepage [${each.value.env}-${each.value.subset}]"
  tags    = ["app:homepage", "env:${each.value.env}", "subset:${each.value.subset}"]
  message = <<-EOT
    runbook: <redacted>
    ${each.value.notify}
    EOT

  options = {
    tick_every           = 3600
    min_failure_duration = 0
    min_location_failed  = 2
    # TODO add retry # of times when this resource supports it
  }
  device_ids = ["chrome.laptop_large", "chrome.mobile_small"]
  locations  = each.value.locations

  request = {
    method = "GET"
    url    = "https://${each.value.subdomain}.<redacted>.com"
  }
  request_headers = {
    accept-encoding = "gzip"
    cookie          = "homepage_subset=${each.value.subset};"
  }

  assertions = [
    # TODO not (yet) supported. must be done in web UI
  ]
}
```

### Expected Behavior
The plan works successfully.

### Actual Behavior
For each browser test, we receive an error:
```
Error: error getting synthetics test:  is not a valid SyntheticsDeviceID: {"status":"live","public_id":"<redacted>","tags":["app:homepage","env:prod","subset:candidate"],"locations":["aws:ap-southeast-2","aws:eu-west-2","aws:us-east-2","aws:us-west-1"],"message":"runbook: <redacted>","name":"homepage [prod-candidate]","monitor_id":<redacted>,"type":"browser","config":{"request":{"url":"<redacted>","headers":{"cookie":"homepage_subset=candidate;","accept-encoding":"gzip"},"method":"GET"}},"options":{"min_failure_duration":0,"device_ids":["chrome.laptop_large","chrome.mobile_small"],"tick_every":3600,"min_location_failed":2}}
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
Only occurs with provider version 2.8.0.

**Comment by zippolyte** [comment]
Hi,

The version 2.8.0 of the provider ships a different library for communicating with Datadog's API. This library has a stricter validation on which values are allowed, but unfortunately, the list wasn't up to date with what's allowed by the API.
We are working on a fix. We'll update this issue accordingly.

Sorry for the inconvenience.

**Comment by gzussa** [comment]
This issue should be fixed with the upcoming 2.9.0 release. See https://github.com/terraform-providers/terraform-provider-datadog/pull/546

**Comment by gzussa** [comment]
This issue should be fixed with the upcoming 2.8.1 release. See https://github.com/terraform-providers/terraform-provider-datadog/pull/546

**Comment by gzussa** [comment]
The release is actually `2.9.0`. See https://github.com/terraform-providers/terraform-provider-datadog/releases/tag/v2.9.0. Closing this issue.
